### DISCUSSION TITLE
Kong API Gateway basic-auth reset consumer password not documented anywhere

### DISCUSSION THREAD

**Discussion by digitalray** [Kong/kong#12596] [Help] [answered]
There is plenty of documentation related on how to create a consumer and a basic-auth credential for it. However, there is no documentation about how to reset a basic-auth password of an existing consumer.

Here is how you can create a consumer from the Kong API Gateway docs:
Create consumer first
```shell
curl -s localhost:8001/consumers -d username=bob | jq
```
Create basic-auth credentials next
```shell
curl -s localhost:8001/consumers/bob/basic-auth -d username=bob -d password=supersecretpassword | jq
```

However, I cannot find anywhere about resetting/changing an existing password.

I have tried the following:
```shell
curl -i -X PUT http://localhost:8001/consumers/devopsadmin/basic-auth \
  --data "username=bob" \
  --data "password=supersecretpassword1"
```
This produces the following error
Status Code: `405`
`{'message': 'Method not allowed'}`

When changing the above curl command to use `PATCH`, I am getting the same error.

Does anyone know how to reset a consumer's basic-auth password in kong without deleting and re-creating the consumer?

**Comment by hanshuebner** [comment, marked as answer]
Hello @digitalray,

you need to specify the ID of the basic-auth credential when updating the password, e.g.

```
$ curl -sX PATCH http://localhost:8001/consumers/bob/basic-auth/a0221fc9-47eb-488a-8f5e-7194d07e08ed -d password=muahaha  | jq .
{
  "id": "a0221fc9-47eb-488a-8f5e-7194d07e08ed",
  "username": "bob",
  "tags": null,
  "password": "1c7eb01cba63eef50eff91cfa522c699b093da73",
  "created_at": 1709648556,
  "consumer": {
    "id": "3a594673-e9c4-4864-9102-5ef0e9abb4e7"
  }
}
```

The documentation is lacking in that respect, I'll open a ticket to fix that.

**Reply by digitalray** [reply]
Thank you @hanshuebner. I'll update the scripts.
